trieste/acquisition/function/active_learning.py

```diff
@@ -245,3 +245,162 @@ def acquisition(x: TensorType) -> TensorType:
+        self._threshold = threshold
+
+    def __repr__(self) -> str:
+        """"""
+        return f"IntegratedVarianceReduction(threshold={self._threshold!r})"
+
+    def prepare_acquisition_function(
+        self,
+        model: ProbabilisticModel,
+        dataset: Optional[Dataset] = None,
+    ) -> AcquisitionFunction:
+        """
+        :param model: The model.
+        :param dataset: Unused.
+
+        :return: The integral of the predictive variance.
+        """
+
+        return integrated_variance_reduction(model, self._integration_points, self._threshold)
+
+    def update_acquisition_function(
+        self,
+        function: AcquisitionFunction,
+        model: ProbabilisticModel,
+        dataset: Optional[Dataset] = None,
+    ) -> AcquisitionFunction:
+        """
+        :param function: The acquisition function to update.
+        :param model: The model.
+        :param dataset: Unused.
+        """
+        return function  # no need to update anything
+
+
+class integrated_variance_reduction(AcquisitionFunctionClass):
+    r"""
+    The reduction of the (weighted) average of the predicted variance over the integration points
+    (a.k.a. Integrated Means Square Error or IMSE criterion).
+    See :cite:`Picheny2010` for details.
+
+    The criterion (to maximise) writes as:
+
+        .. math:: \int_x (v_{old}(x) - v_{new}(x)) * weights(x),
+
+    where :math:`v_{old}(x)` is the predictive variance of the model at :math:`x`, and
+    :math:`v_{new}(x)` is the updated predictive variance, given that the GP is further
+    conditioned on the query points.
+
+    Note that since :math:`v_{old}(x)` is constant w.r.t. the query points, this function
+    only returns :math:`-\int_x v_{new}(x) * weights(x)`.
+
+    If no threshold is provided, the goal is to learn a globally accurate model, and
+    the predictive variance (:math:`v_{new}`) is used. Otherwise, learning is 'targeted'
+    towards regions where the GP is close to particular values, and the variance is weighted
+    by the posterior GP pdf evaluated at the threshold T (if a single value is given) or by the
+    probability that the GP posterior belongs to the interval between the 2 thresholds T1 and T2
+    (note the slightly different parametrisation compared to :cite:`Picheny2010` in that case).
+
+    This criterion allows batch size > 1. Note that the computational cost grows cubically with
+    the batch size.
+
+    This criterion requires a method (conditional_predict_f) to compute the new predictive variance
+    given that query points are added to the data.
+    """
+
+    def __init__(
+        self,
+        model: ProbabilisticModel,
+        integration_points: TensorType,
+        threshold: Optional[Sequence[float]] = None,
+    ):
+        """
+        :param model: The model of the objective function.
+        :param integration_points: points over which to integrate the objective prediction variance
```

Comment:
perhaps we should add a recommendation for this, sample of points that cover the search space well enough?
```suggestion
        :param integration_points: Points over which to integrate the objective prediction variance.
```